woodwork/logical_types.py

```diff
@@ -173,21 +178,22 @@ def __init__(self, datetime_format=None):
         self.datetime_format = datetime_format
 
     def transform(self, series):
-        """Converts the series data to a formatted datetime."""
+        """Converts the series data to a formatted datetime. Datetime format will be inferred if datetime_format is None."""
         new_dtype = self._get_valid_dtype(type(series))
         if new_dtype != str(series.dtype):
+            self.datetime_format = self.datetime_format or _infer_datetime_format(series)
             try:
                 if dd and isinstance(series, dd.Series):
                     name = series.name
-                    series = dd.to_datetime(series, format=self.datetime_format, infer_datetime_format=True)
+                    series = dd.to_datetime(series, format=self.datetime_format)
                     series.name = name
                 elif ks and isinstance(series, ks.Series):
                     series = ks.Series(ks.to_datetime(series.to_numpy(),
                                                       format=self.datetime_format,
                                                       infer_datetime_format=True),
```

Comment:
I noticed we remove `infer_datetime_format=True` in `dd.to_datetime` and `pd.to_datetime`. Do we also want to remove `infer_datetime_format=True` in `ks.to_datetime` or is this needed?

Reply:
nope, it's not. good catch!